### PR TITLE
[FIX] [PISA25-269] endpoint update metada is not indexing metadata on advanced search

### PR DESCRIPTION
**Pull request by andreluizmachado**
Ticket: https://oat-sa.atlassian.net/browse/PISA25-269

## What's Changed

- fix: Replaced returnSuccess by setSuccessJsonResponse on updateMetadata action
- fix: version attribute removed of the success response of the endpoint updateMetadata

## How to test
- Please check ticket https://oat-sa.atlassian.net/browse/PISA25-269

### CODE REVIEW

File: controller/AbstractRestQti.php [hunk 1]
```diff
@@ -219,7 +221,7 @@ public function updateMetadata(): void
             throw new BadRequestException($exception->getMessage(), 422);
         }
 
-        $this->returnSuccess($metadataInput);
+        $this->setSuccessJsonResponse($metadataInput);
```
Comment (Karol-Stelmaczonek) [inline controller/AbstractRestQti.php:224]: Ok I found one thing, its abstract class so possible many extensions of it. Are we sure in every place we want to continue with those event you @andreluizmachado showed us on meeting? Maybe somewhere is some piece of code whitch depends on this removed exit() ? Can we somehow check it?

**Review by Karol-Stelmaczonek — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: controller/AbstractRestQti.php (hunk 1 above)
Comment (andreluizmachado) [reply controller/AbstractRestQti.php:224]: Very good Question @Karol-Stelmaczonek. But it should be ok, actually, this implementation is quite new and for sure that events are mandatory to be thrown anywhere.

**Review by pnal — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**